### PR TITLE
[W6.1d][T16-B2]Lin Jiachao

### PR DESCRIPTION
**Pull request by lincredibleJC**
Add a polymorphic isMutating method

### CODE REVIEW

File: src/seedu/addressbook/commands/Command.java [hunk 1]
```diff
@@ -70,4 +70,8 @@ public int getTargetIndex() {
     public void setTargetIndex(int targetIndex) {
         this.targetIndex = targetIndex;
     }
+
+    public boolean isMutating() {
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/commands/Command.java:74]: Command#isMutating() does not have a header comment.
This method should have a header comment. If not, users of the method as well as those who overrides method will not know how exactly the method behaves.

File: src/seedu/addressbook/logic/Logic.java [hunk 2]
```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()){
```
Comment (brandonyeoxg) [inline src/seedu/addressbook/logic/Logic.java:88]: [Coding standard](https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html) violation! One more more of these problems detected:
1. missing space after `(command.isMutating())`

File: test/java/seedu/addressbook/logic/LogicTest.java [hunk 3]
```diff
@@ -90,7 +90,10 @@ private void assertCommandBehavior(String inputCommand,
         //Confirm the state of data is as expected
         assertEquals(expectedAddressBook, addressBook);
         assertEquals(lastShownList, logic.getLastShownList());
-        assertEquals(addressBook, saveFile.load());
+        if(inputCommand.equals("add")||inputCommand.equals("clear")||inputCommand.equals("delete")){
```
Comment (brandonyeoxg) [inline test/java/seedu/addressbook/logic/LogicTest.java:93]: Perhaps another solution would be to make use of the `isMutating()` that you have created.

```java
if(new Parser().parseCommand(inputCommand).isMutating()) {
    assertEquals(addressBook, saveFile.load());
}
```

In this way you ensure that when a new mutating command is added, you can be sure that you are testing those new commands.

**Review by brandonyeoxg — COMMENTED** [review]
@lincredibleJC Some comments added. Please close the PR after reading comments Good effort keep it up!

File: src/seedu/addressbook/commands/Command.java (hunk 1 above)
Comment (lincredibleJC) [reply src/seedu/addressbook/commands/Command.java:74]: So something like this?
/**
     * Returns a boolean to indicate whether the command mutates the stored addressBook data
     *
     * @return boolean from overriding method in subclass, false if subclass does not have an overriding method
     */
    public boolean isMutating() {
        return false;
    }

**Pull request closed without merging**